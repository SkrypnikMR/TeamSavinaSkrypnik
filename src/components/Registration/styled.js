import styled from 'styled-components';

export const StRegDiv = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
`;
