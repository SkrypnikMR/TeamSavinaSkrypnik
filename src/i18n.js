import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import en from './locales/en.json';
import ru from './locales/ru.json';

const language = localStorage.getItem('lang') || 'en';

i18n.use(initReactI18next).init({
    resources: {
        en: { translations: en },
        ru: { translations: ru },
    },
    fallbackLng: ['en', 'ru'],
    lng: language,
    debug: false,
    ns: ['translations'],
    defaultNS: 'translations',
    keySeparator: false,
    interpolation: {
        escapeValue: false,
        formatSeparator: ',',
    },
    react: {
        wait: true,
    },
});

export default i18n;
