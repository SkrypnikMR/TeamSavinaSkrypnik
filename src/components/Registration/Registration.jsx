import React from 'react';
import PropTypes from 'prop-types';
import { regInputs } from '/src/constants/componentsÐ¡onsts';
import { NavLink, Redirect } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { APP_ROUTES } from '/src/constants/reactRoutes';
import { StRegDiv } from './styled';
import Form from '../UI/Form';
import Input from '../UI/Input';
import Button from '../UI/Button';

const Registration = ({ sendRegistrationRequest, setRegistrationValue, fields }) => {
    const { t } = useTranslation();
    const { success } = fields;
    if (success) return <Redirect to={APP_ROUTES.login} />;
    const handleOnChange = (data) => {
        const { error } = fields;
        if (error) setRegistrationValue({ name: 'error', value: null });
        setRegistrationValue(data);
    };
    return (
        <StRegDiv>
            <Form>
                <p>{t('registration')}</p>
                {regInputs.map(input => (
                    <Input
                        width="80%"
                        id={input.id}
                        height="100px"
                        key={input.id}
                        name={input.id}
                        inputHeight="50px"
                        borderRadius="5px"
                        label={t(input.label)}
                        value={fields[input.id]}
                        onChange={handleOnChange}
                        placeholder={t(input.placeholder)}
                    />
                ))}
                <Button
                    height="50px"
                    content="reg_btn"
                    id="registration"
                    onClick={sendRegistrationRequest}
                />
                <span>
                    {t('reg_span_text')}
                    <NavLink to={APP_ROUTES.login}>{t('reg_link_text')}</NavLink>
                </span>
            </Form>
        </StRegDiv>
    );
};

Registration.propTypes = {
    sendRegistrationRequest: PropTypes.func.isRequired,
    setRegistrationValue: PropTypes.func.isRequired,
    fields: PropTypes.object.isRequired,
};

export default Registration;
