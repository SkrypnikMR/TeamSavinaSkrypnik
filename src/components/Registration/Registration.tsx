import React from 'react';
import PropTypes from 'prop-types';
import { NavLink, Redirect } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import { regInputs } from 'src/constants/componentsÐ¡onsts';
import { APP_ROUTES } from '../../constants/reactRoutes';
import Form from '../UI/Form';
import Input from '../UI/Input';
import Button from '../UI/Button';
import { useTheme } from '../Hook/useTheme';

import { StRegDiv } from './styled';

const Registration = ({ sendRegistrationRequest, setRegistrationValue, fields }) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();
    const { success } = fields;
    if (success) return <Redirect to={APP_ROUTES.login} />;
    const handleOnChange = (data) => {
        const { error } = fields;
        if (error) setRegistrationValue({ name: 'error', value: null });
        setRegistrationValue(data);
    };

    return (
        <StRegDiv colors={colors} theme={theme}>
            <img src="../../../public/assets/images/game-development.png"/>
            <Form>
                <p>{t('registration')}</p>
                {regInputs.map((input) => (
                    <Input
                        width="80%"
                        id={input.id}
                        height="100px"
                        key={input.id}
                        name={input.id}
                        inputHeight="50px"
                        borderRadius="5px"
                        label={t(input.label)}
                        value={fields[input.id]}
                        onChange={handleOnChange}
                        placeholder={t(input.placeholder)}
                    />
                ))}
                <Button
                    height="50px"
                    content="reg_btn"
                    id="registration"
                    onClick={sendRegistrationRequest}
                    focusColor="black"
                    name=""
                    value=""
                    type="button"
                    title=""
                />
                <span>
                    {t('reg_span_text')}
                    <NavLink to={APP_ROUTES.login}>{t('reg_link_text')}</NavLink>
                </span>
            </Form>
        </StRegDiv>
    );
};

Registration.propTypes = {
    sendRegistrationRequest: PropTypes.func.isRequired,
    setRegistrationValue: PropTypes.func.isRequired,
    fields: PropTypes.object.isRequired,
};

export default Registration;
