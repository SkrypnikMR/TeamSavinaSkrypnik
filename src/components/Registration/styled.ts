import styled from 'styled-components';

export const StRegDiv = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    background: ${({ colors, theme }) => colors[theme].backgroundLogReg};
    height: 100vh;
    img{
        height: 40vh;
    }
`;
