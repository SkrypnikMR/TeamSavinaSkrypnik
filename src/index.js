import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import rootReducer from './Store';
import App from './components/App';
import rootSaga from './Store/saga';
import './i18n';

const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(saga),
    devTools,
  ),
);
saga.run(rootSaga);

render(
  <Provider store={store}>
    <App />
  </Provider >,
  document.getElementById('root'),
);
