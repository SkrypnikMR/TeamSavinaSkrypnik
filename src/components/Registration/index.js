import { connect } from 'react-redux';
import Registration from './Registration.jsx';
import { setRegistrationValue, sendRegistrationRequest } from '/src/Store/registration/actions';
import { registrationStore } from '/src/Store/registration/selectors';

const mapStateToProps = state => ({ fields: registrationStore(state) });

const mapDispatchToProps = dispatch => ({
    setRegistrationValue: payload => dispatch(setRegistrationValue(payload)),
    sendRegistrationRequest: payload => dispatch(sendRegistrationRequest(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Registration);
