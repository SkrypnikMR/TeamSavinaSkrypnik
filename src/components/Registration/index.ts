import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { setRegistrationValue, sendRegistrationRequest } from '../../store/registration/actions';
import { registrationStore } from '../../store/registration/selectors';
import Registration from './Registration';
import { ApplicationState } from '../../store/types';
import { TRegistartionPayload } from '../../store/registration/types';

const mapStateToProps = (state: ApplicationState) => ({ fields: registrationStore(state) });

const mapDispatchToProps = (dispatch: Dispatch) => ({
    setRegistrationValue: (payload: TRegistartionPayload) => dispatch(setRegistrationValue(payload)),
    sendRegistrationRequest: () => dispatch(sendRegistrationRequest()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Registration);
