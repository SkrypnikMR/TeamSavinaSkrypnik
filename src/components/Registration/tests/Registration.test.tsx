import React from 'react';
import Registration from '../Registration';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';
import 'react-router-dom';
import '/src/i18n';

describe('Registration', () => {
    let props;
    const sendRegistrationRequest = jest.fn();
    const setRegistrationValue = jest.fn();
    const fields = {
        login: '',
        password: '',
        confrim: '',
        success: false,
        error: false,
    };
    beforeEach(() => {
        props = {
            setRegistrationValue,
            sendRegistrationRequest,
            fields,
        };
    });
    it('Should match snapshot', () => {
        const component = shallowSmart(<Registration {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = mountSmart(<Registration {...props} />);
        expect(component.find('p')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = mountSmart(<Registration {...props} />);
        expect(component.find('Input')).toHaveLength(3);
    });
    it('should render button', () => {
        const component = mountSmart(<Registration {...props} />);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should click on the button', () => {
        const component = mountSmart(<Registration {...props} />);
        component.find('Button').simulate('click');
        expect(sendRegistrationRequest).toHaveBeenCalled();
    });
    it('should click change input', () => {
        const component = mountSmart(<Registration {...props} />);
        component.find('input').at(0).simulate('change', { target: { name: 'login', value: 'emailValue' } });
        expect(setRegistrationValue).toHaveBeenCalledWith({ name: 'login', value: 'emailValue' });
    });
    it('should click change input with error', () => {
        props.fields = {
            error: true,
        };
        const component = mountSmart(<Registration {...props} />);
        component.find('input').at(0).simulate('change', { target: { name: 'email', value: 'emailValue' } });
        expect(setRegistrationValue).toHaveBeenCalledWith({ name: 'email', value: 'emailValue' });
    });
    it('should redirect', () => {
        props.fields = {
            success: true,
        };
        const component = mountSmart(<Registration {...props} />);
        expect(component.find('Redirect')).toHaveLength(1);
    });
});
