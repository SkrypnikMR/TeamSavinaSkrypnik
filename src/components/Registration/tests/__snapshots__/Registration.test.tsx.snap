// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Should match snapshot 1`] = `"<div class=\\"styled__StRegDiv-sc-wqf9bf-0 hrjOKK\\"><img src=\\"../../../public/assets/images/game-development.png\\"/><div class=\\"styled__StFormDiv-sc-p60jdb-0 fMNLMW\\"><p>Sign Up</p><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-sc-e64755-3 bIHjkn\\"><label for=\\"login\\" class=\\"styled__StLabel-sc-e64755-0 kqGJQd\\">Login</label><input type=\\"text\\" id=\\"login\\" name=\\"login\\" placeholder=\\"Enter your login...\\" class=\\"styled__StInput-sc-e64755-1 kmoiCN\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-sc-e64755-3 bIHjkn\\"><label for=\\"password\\" class=\\"styled__StLabel-sc-e64755-0 kqGJQd\\">Password</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" placeholder=\\"Enter your password...\\" class=\\"styled__StInput-sc-e64755-1 kmoiCN\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-sc-e64755-3 bIHjkn\\"><label for=\\"confirm\\" class=\\"styled__StLabel-sc-e64755-0 kqGJQd\\">Confirm</label><input type=\\"text\\" id=\\"confirm\\" name=\\"confirm\\" placeholder=\\"Confirm your password...\\" class=\\"styled__StInput-sc-e64755-1 kmoiCN\\"/></div><button id=\\"registration\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-1y35zax-0 fRKlXt\\">Register</button><span>Alredy Registred?<a aria-current=\\"page\\" class=\\"active\\" href=\\"/\\">Sign In</a></span></div></div>"`;
